src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -362,6 +362,12 @@ private JpegColorSpace DeduceJpegColorSpace(byte componentCount)
                     return JpegColorSpace.RGB;
                 }
 
+                // If the component Id's are R, G, B in ASCII the colorspace is RGB and not YCbCr.
+                if (components[0].Id == 82 && components[1].Id == 71 && components[2].Id == 66)
```

Comment:
If you switch the order of these checks I believe we avoid the bounds checks for 2 out of 3 of them.